estimagic/optimization/reparametrize.py

```diff
@@ -78,8 +83,10 @@ def reparametrize_from_internal(internal_params, constraints, original_params):
 
     for constr in constraints:
         params_subset = reindexed.loc[constr["index"]]
-        if constr["type"] == "covariance":
-            params_sr.update(_covariance_from_internal(params_subset, constr["case"]))
+        if constr["type"] in ["covariance", "sdcorr"]:
```

Comment:
There is no else. I assume there are other checks before which would capture this, but maybe you want to add it so that you are sure. maybe ``else: raise NotImplementedError(f"Constraint {constr['type']} is not implemented")``

Reply:
Seconded. I think you want to handle the entire block for the constraints (lines 32-53 in 647a18d) in one go rather than three steps for this very reason. Very easy to miss a mistyped constraint (yes, you probably *want* to check that elsewhere...) or so and very hard to debug. 

Similarly, lines 55-63 are unnecessarily hard to understand IMHO. Why not something like (untested):
```
internal = internal.query("(fixed == False) | (lower == upper)").drop(["__fixed__"], axis=1)
```

Btw, `_fixed` instead of `__fixed__` would look more pythonic to me

Reply:
Thanks for your feedback. I'm currently working on this function and will try to make it more readable. Also, I will add an else statement. But it is not possible to handle all types of constraints in one go. It is important that fixes and equality constraints are applied before all other constraints because:

- In some cases they determine which transformation is applied to the parameters
- We have to check if fixes and equality constraints are compatible with the other constraints. 

I'll ask you for a second round of feedback on this function once I'm done.

Reply:
> But it is not possible to handle all types of constraints in one go. It is important that fixes and equality constraints are applied before all other constraints

How about a custom sort of the list of constraints?

Reply:
Sorting the constraints and then running them through one block feels too implicit. Due to some restructuring I already reduced the dependencies between the constraints. Now only fixes have to be applied before equality constraints, otherwise the order is irrelevant. I propose the following:

instead of a column  `'_fixed'` in params, I keep this information in a separate Series called `fixed`. This will be an argument to some of the other constraint functions and thereby clearly shows the dependency. All other constraints will be handled in one block and I will have an else statement in the end. 

I think this dependency was also the reason why I originally wanted the user to specify fixes directly in params and not in constraints, but while using estimagic in skillmodels I discovered that specifying them in constraints is much more convenient. 

This solution would also allow for more code sharing between the application of the constraints and the processing and checking of the user provided constraints. 

@hmgaudecker maybe we can briefly discuss the details in the afternoon.